openmc/data/kalbach_mann.py

```diff
@@ -0,0 +1,402 @@
+                    interpolation[i] = 1 if eout.interpolation == 'histogram' else 2
+                elif isinstance(eout, Discrete):
+                    n_discrete_lines[i] = n
+                    interpolation[i] = 1
+                distribution[0, j:j+n] = eout.x
+                distribution[1, j:j+n] = eout.p
+                distribution[2, j:j+n] = eout.c
+
+            distribution[3, j:j+n] = km_r.y
+            distribution[4, j:j+n] = km_a.y
+            j += n
+
+        # Create dataset for distributions
+        dset = group.create_dataset('distribution', data=distribution)
+
+        # Write interpolation as attribute
+        dset.attrs['offsets'] = offsets
+        dset.attrs['interpolation'] = interpolation
+        dset.attrs['n_discrete_lines'] = n_discrete_lines
+
+    @classmethod
+    def from_hdf5(cls, group):
+        """Generate Kalbach-Mann distribution from HDF5 data
+
+        Parameters
+        ----------
+        group : h5py.Group
+            HDF5 group to read from
+
+        Returns
+        -------
+        openmc.data.KalbachMann
+            Kalbach-Mann energy distribution
+
+        """
+        interp_data = group['energy'].attrs['interpolation']
+        energy_breakpoints = interp_data[0, :]
+        energy_interpolation = interp_data[1, :]
+        energy = group['energy'][()]
+
+        data = group['distribution']
+        offsets = data.attrs['offsets']
+        interpolation = data.attrs['interpolation']
+        n_discrete_lines = data.attrs['n_discrete_lines']
+
+        energy_out = []
+        precompound = []
+        slope = []
+        n_energy = len(energy)
+        for i in range(n_energy):
+            # Determine length of outgoing energy distribution and number of
+            # discrete lines
+            j = offsets[i]
+            if i < n_energy - 1:
+                n = offsets[i+1] - j
+            else:
+                n = data.shape[1] - j
+            m = n_discrete_lines[i]
+
+            # Create discrete distribution if lines are present
+            if m > 0:
+                eout_discrete = Discrete(data[0, j:j+m], data[1, j:j+m])
+                eout_discrete.c = data[2, j:j+m]
+                p_discrete = eout_discrete.c[-1]
+
+            # Create continuous distribution
+            if m < n:
+                interp = INTERPOLATION_SCHEME[interpolation[i]]
+                eout_continuous = Tabular(data[0, j+m:j+n], data[1, j+m:j+n], interp)
+                eout_continuous.c = data[2, j+m:j+n]
+
+            # If both continuous and discrete are present, create a mixture
+            # distribution
+            if m == 0:
+                eout_i = eout_continuous
+            elif m == n:
+                eout_i = eout_discrete
+            else:
+                eout_i = Mixture([p_discrete, 1. - p_discrete],
+                                 [eout_discrete, eout_continuous])
+
+            km_r = Tabulated1D(data[0, j:j+n], data[3, j:j+n])
```

Comment:
It is unclear to me why `km_r` and `km_a` are the same data points except 3 vs 4. Could a comment be added to explain this?